packages/app-store/exchangecalendar/lib/CalendarService.ts

```diff
@@ -0,0 +1,204 @@
+    return appointment.Delete(DeleteMode.MoveToDeletedItems).catch((reason) => {
+      this.log.error(reason);
+      throw reason;
+    });
+  }
+
+  async getAvailability(
+    dateFrom: string,
+    dateTo: string,
+    selectedCalendars: IntegrationCalendar[]
+  ): Promise<EventBusyDate[]> {
+    const calendars: IntegrationCalendar[] = await this.listCalendars();
+    const promises: Promise<EventBusyDate[]>[] = calendars
+      .filter((lcal) => selectedCalendars.some((rcal) => lcal.externalId == rcal.externalId))
+      .map(async (calendar) => {
+        return this.getExchangeService()
+          .FindAppointments(
+            new FolderId(calendar.externalId),
+            new CalendarView(DateTime.Parse(dateFrom), DateTime.Parse(dateTo))
+          )
+          .then((results: FindItemsResults<Appointment>) => {
+            return results.Items.filter((appointment: Appointment) => {
+              return appointment.LegacyFreeBusyStatus != LegacyFreeBusyStatus.Free;
+            }).map((appointment: Appointment) => {
+              return {
+                start: new Date(appointment.Start.ToISOString()),
+                end: new Date(appointment.End.ToISOString()),
+              };
+            });
+          })
+          .catch((reason) => {
+            this.log.error(reason);
+            throw reason;
+          });
+      });
+    return Promise.all(promises).then((x) => x.flat());
+  }
+
+  async listCalendars(): Promise<IntegrationCalendar[]> {
+    const service: ExchangeService = this.getExchangeService();
+    const view: FolderView = new FolderView(1000);
+    view.PropertySet = new PropertySet(BasePropertySet.IdOnly);
+    view.PropertySet.Add(FolderSchema.ParentFolderId);
+    view.PropertySet.Add(FolderSchema.DisplayName);
+    view.PropertySet.Add(FolderSchema.ChildFolderCount);
+    view.Traversal = FolderTraversal.Deep;
+    const deletedItemsFolder: Folder = await Folder.Bind(service, WellKnownFolderName.DeletedItems);
+    const searchFilterCollection = new SearchFilter.SearchFilterCollection(LogicalOperator.And);
+    searchFilterCollection.Add(new SearchFilter.IsEqualTo(FolderSchema.FolderClass, "IPF.Appointment"));
+    return service
+      .FindFolders(WellKnownFolderName.MsgFolderRoot, searchFilterCollection, view)
+      .then((res: FindFoldersResults) => {
+        return res.Folders.filter((folder: Folder) => {
+          return folder.ParentFolderId.UniqueId != deletedItemsFolder.Id.UniqueId;
+        }).map((folder: Folder) => {
+          return {
+            externalId: folder.Id.UniqueId,
+            name: folder.DisplayName ?? "",
+            primary: folder.ChildFolderCount > 0,
+            integration: this.integrationName,
+          };
+        });
+      })
+      .catch((reason) => {
+        this.log.error(reason);
+        throw reason;
+      });
+  }
+
+  private getExchangeService(): ExchangeService {
+    const service: ExchangeService = new ExchangeService();
```

Comment:
Looking at the `ews-javascript-api` docs I always thought you had to pass the Exchange version. I decided to dive deeper into the actual EWS docs and saw that you don't have to specify and it will use the latest version of EWS which supports 2013 and 2016. I feel good about this PR but will want to see what others will have to say.

ews-javascript-api docs: https://github.com/gautamsi/ews-javascript-apigetExchangeService
EWS ExchangeService class constructors: https://docs.microsoft.com/en-us/dotnet/api/microsoft.exchange.webservices.data.exchangeservice?view=exchange-ews-api#constructors

Reply:
I really didn't expect that nobody is using the Exchange integration. At least here in Germany, it feels like any organization is using Exchange. But maybe I'm biased because I'm connected to like 12 Exchange calendars 😃